x-pack/plugins/security/server/routes/analytics/record_violations.ts

```diff
@@ -0,0 +1,198 @@
+         */
+        sourceFile: schema.maybe(schema.string()),
+        /**
+         * If known, the line number in sourceFile where the violation occured, or null otherwise.
+         */
+        lineNumber: schema.maybe(schema.number()),
+        /**
+         * If known, the column number in sourceFile where the violation occured, or null otherwise.
+         */
+        columnNumber: schema.maybe(schema.number()),
+        /**
+         * A string indicating whether the violated permissions policy was enforced in this case. disposition will be set to "enforce" if the policy was enforced, or "report" if the violation resulted only in this report being generated (with no further action taken by the user agent in response to the violation).
+         */
+        disposition: schema.oneOf([schema.literal('enforce'), schema.literal('report')]),
+      },
+      { unknowns: 'ignore' }
+    ),
+  },
+  { unknowns: 'ignore' }
+);
+
+/**
+ * Interface that represents a permissions policy violation report according to W3C spec.
+ */
+export type PermissionsPolicyViolationReport = TypeOf<
+  typeof permissionsPolicyViolationReportSchema
+>;
+
+/**
+ * This endpoint receives reports from the user's browser via the Reporting API when one of our
+ * `Content-Security-Policy` or `Permissions-Policy` directives have been violated.
+ */
+export function defineRecordViolations({ router, analyticsService }: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/analytics/_record_violations',
+      validate: {
+        /**
+         * Chrome supports CSP3 spec and sends an array of reports. Safari only sends a single
+         * report so catering for both here.
+         */
+        body: schema.oneOf([
+          schema.arrayOf(
+            schema.oneOf([cspViolationReportSchema, permissionsPolicyViolationReportSchema])
+          ),
+          cspViolationReportSchema,
+        ]),
+      },
+      options: {
+        /**
+         * Browsers will stop sending reports for the duration of the browser session and without
+         * further retries once this endpoint has returned a single 403. This would effectively
+         * prevent us from capture any reports. To work around this behaviour we optionally
+         * authenticate users but silently ignore any reports that have been received from
+         * unauthenticated users.
+         */
+        authRequired: 'optional',
+        /**
+         * This endpoint is called by the browser in the background so `kbn-xsrf` header is not sent.
+         */
+        xsrfRequired: false,
+        access: 'internal',
+        body: {
+          /**
+           * Both `application/reports+json` (CSP3 spec) and `application/csp-report` (Safari) are
+           * valid values but Hapi does not parse the request body when `application/csp-report` is
+           * specified so enforcing JSON mime-type for this endpoint.
+           */
+          override: 'application/json',
```

Comment:
(following previous comment) ok, I get what you needed that for.

I wonder if the correct solution wouldn't be to register the missing mimetype, (as in https://github.com/elastic/kibana/pull/97823/files for example)

but adding this `override` body option is probably fine too.

Reply:
Ah, that could be an option as well. How does the server options `mime` override work? Does it mean that all routes would start accepting `application/csp-report` since it gets converted to `application/json` before it hits the route validator?  I think what's nice about the `override` option is that you can define it at a route level so we're not changing global behaviour.